drop-transfer/src/file/mod.rs

```diff
@@ -296,19 +296,27 @@ impl FileToSend {
     }
 
     /// Calculate sha2 of a file. This is a blocking operation
-    pub(crate) async fn checksum(&self, limit: u64) -> crate::Result<[u8; 32]> {
+    pub(crate) async fn checksum(
+        &self,
+        limit: u64,
+        progress_tx: Option<tokio::sync::watch::Sender<u64>>,
+    ) -> crate::Result<[u8; 32]> {
         let reader = reader::open(&self.source)?;
         let mut reader = io::BufReader::new(reader).take(limit);
-        let csum = checksum(&mut reader).await?;
+        let csum = checksum(&mut reader, progress_tx).await?;
         Ok(csum)
     }
 }
 
-pub async fn checksum(reader: &mut impl io::Read) -> io::Result<[u8; 32]> {
+pub async fn checksum(
+    reader: &mut impl io::Read,
+    progress_tx: Option<tokio::sync::watch::Sender<u64>>,
```

Comment:
IMHO it should be more generic and take a progress callback, e.g.
```rust
fn checksum(reader: &mut ..., progress_cb: impl Fn(u64))
```

Reply:
I actually went the callback route initially but that deemed to be very difficult(to me) as the callback needs to issue async operations, thus I opted out for a channel instead.

Reply:
Additional commit contains the conversion to a callback. Actually I've had success, even though it's was much more annoying to implement